common/errors/error.go

```diff
@@ -33,21 +36,26 @@ func (s *ServerError) Model() models.ServerError {
 	}
 }
 
-func NewEntityDoesNotExistError(entityType, entityID string) *ServerError {
-	switch entityType {
-	case "deploy":
-		return Newf(DeployDoesNotExist, "Deploy '%s' does not exist", entityID)
-	case "environment":
-		return Newf(EnvironmentDoesNotExist, "Environment '%s' does not exist", entityID)
-	case "job":
-		return Newf(JobDoesNotExist, "Job '%s' does not exist", entityID)
-	case "load_balancer":
-		return Newf(LoadBalancerDoesNotExist, "Load balancer '%s' does not exist", entityID)
-	case "service":
-		return Newf(ServiceDoesNotExist, "Service '%s' does not exist", entityID)
-	case "task":
-		return Newf(TaskDoesNotExist, "Task '%s' does not exist", entityID)
+func (s *ServerError) Write(w http.ResponseWriter, r *http.Request) {
+	switch s.Code {
+	case InvalidRequest,
```

Comment:
[nit] I would expect InvalidRequest and DoesNotExist errors to return different http status codes. 400 & 404 respectively.

Reply:
addressed above